udemy_exercises/songs/Album.java

```diff
@@ -1,19 +1,54 @@
-package udemy_exercises.songs;
+package com.adamrice;
 
-class Album {
+import java.util.ArrayList;
+import java.util.LinkedList;
 
-}
+public class Album {
+    private String name;
+    private String artist;
+    private ArrayList<Song> songs;
+
+    public Album(String name, String artist) {
+        this.name = name;
+        this.artist = artist;
+        this.songs = new ArrayList<Song>();
+    }
+
+    public boolean addSong(String title, double duration) {
+        if (findSong(title) == null) {
+            this.songs.add(new Song(title, duration));
+            return true;
+        }
+        return false;
+    }
+
+    private Song findSong(String title) {
+        for (Song checkedSong : this.songs) {
+            if (checkedSong.getTitle().equals(title)) {
+                return checkedSong;
+            }
+        }
+        return null;
+    }
 
-/*
-**Song Player w/ Java Classes**
-     - **Album class** 
-       - Playlist of **Songs**.
-       - addPlayist() (songs must appear in the list in the order by which they were added)
-       - stop()
-       - nextSong()
-       - previousSong()
-       - repeatSong()
-       - showPlaylist()
-       - removePlaylist()
-     - **A song must exist in an album before it can be added to the playlist (so you can only play songs that you own.)**
-*/
\ No newline at end of file
+    public boolean addToPlayList(int trackNumber, LinkedList<Song> playList) {
+        int index = trackNumber - 1;
+        if ((index > 0) && (index <= this.songs.size())) {
```

Comment:
Couple of issues with this 'If' condition:
 - when trackNumber is '1', then index is calculated as '0', which means the first track will never be added, and return 'false', making the client believe there was an error condition
 - when trackNumber is provided that sets 'index' equal to the size of the playList (e.g. for a list of 6 songs, someone enters #7, which is non-existent, and the index then gets set to '6'), then line 17 will attempt to get the 7th item from the list using index '6', and throw an exception.
The test for a valid index needs to be fixed.  I would also recommend encapsulating this test in a way that clearly indicates the index number is being validated.